### PR TITLE
add generated_reviews_enth

### PR DESCRIPTION
**Pull request by cstorm125**
`generated_reviews_enth` is created as part of [scb-mt-en-th-2020](https://arxiv.org/pdf/2007.03541.pdf) for machine translation task. This dataset (referred to as `generated_reviews_yn` in [scb-mt-en-th-2020](https://arxiv.org/pdf/2007.03541.pdf)) are English product reviews generated by [CTRL](https://arxiv.org/abs/1909.05858), translated by Google Translate API and annotated as accepted or rejected (`correct`) based on fluency and adequacy of the translation by human annotators. This allows it to be used for English-to-Thai translation quality esitmation (binary label), machine translation, and sentiment analysis.

### CODE REVIEW

File: datasets/generated_reviews_enth/generated_reviews_enth.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        ),
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "en_segment": datasets.Value("string"),
+                    "th_segment": datasets.Value("string"),
```
Comment (lhoestq) [inline datasets/generated_reviews_enth/generated_reviews_enth.py:65]: Maybe you can use the `Translation` feature type ?
see [flores](https://github.com/huggingface/datasets/blob/master/datasets/flores/flores.py) for an example of usage

**Review by lhoestq — COMMENTED** [review]
Awesome nice job !
thank you

I left one comment about the feature types. Usually for translation we use the `Translation` feature type, could you do this change please ?

File: datasets/generated_reviews_enth/generated_reviews_enth.py (hunk 1 above)
Comment (cstorm125) [reply datasets/generated_reviews_enth/generated_reviews_enth.py]: Got it. Changed.

File: datasets/generated_reviews_enth/generated_reviews_enth.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+
+class GeneratedReviewsEnth(datasets.GeneratorBasedBuilder):
+    """`generated_reviews_enth`: Generated product reviews dataset for machine translation quality prediction, part of [scb-mt-en-th-2020](https://arxiv.org/pdf/2007.03541.pdf)"""
+
+    _DOWNLOAD_URL = "https://github.com/vistec-AI/generated_reviews_enth/raw/main/data.zip"
+    _TRAIN_FILE = "train.jsonl"
+    _VAL_FILE = "valid.jsonl"
+    _TEST_FILE = "test.jsonl"
+
```
Comment (lhoestq) [inline datasets/generated_reviews_enth/generated_reviews_enth.py:51]:
```suggestion
    BUILDER_CONFIG_CLASS = GeneratedReviewsEnthConfig
```
we need this to specify that we need a custom config

**Review by lhoestq — APPROVED** [review]
Nice thanks for doing this change :)

**Pull request merged**